dll/cpl/sysdm/advanced.c

```diff
@@ -11,76 +11,153 @@
 #include "precomp.h"
 
 static TCHAR BugLink[] = _T("http://jira.reactos.org/");
-static TCHAR ReportAsWorkstationKey[] = _T("SYSTEM\\CurrentControlSet\\Control\\ReactOS\\Settings\\Version");
 
+typedef enum _NT_PRODUCT_TYPE
+{
+    NtProductWinNt = 1,
+    NtProductLanManNt,
+    NtProductServer
+} NT_PRODUCT_TYPE, *PNT_PRODUCT_TYPE;
 
-static VOID
-OnOK(HWND hwndDlg)
+static const WCHAR ProductOptions[] = L"System\\CurrentControlSet\\Control\\ProductOptions";
+
+static BOOL
+DoGetProductType(PNT_PRODUCT_TYPE ProductType)
 {
     HKEY hKey;
-    DWORD ReportAsWorkstation;
-
-    ReportAsWorkstation = (SendDlgItemMessageW(hwndDlg,
-                                               IDC_REPORTASWORKSTATION,
-                                               BM_GETCHECK,
-                                               0,
-                                               0) == BST_CHECKED);
-
-    if (RegCreateKeyEx(HKEY_LOCAL_MACHINE,
-                       ReportAsWorkstationKey,
-                       0,
-                       NULL,
-                       0,
-                       KEY_WRITE,
-                       NULL,
-                       &hKey,
-                       NULL) == ERROR_SUCCESS)
+    LONG error;
+    WCHAR szValue[32];
```

Comment:
```suggestion
    WCHAR szValue[32] = {0};
```

Reply:
OK but I don't agree because it slows down the speed.